### PR TITLE
fix default_config after KMS

### PR DESCRIPTION
**Pull request by motobob**
Problem:
after changes for KMS default config does not extract credentials from metadata when running in on EC2 instance

Solution:
Allow to extract them

### CODE REVIEW

**Comment by motobob** [comment]
@kkuzmin

**Comment by kkuzmin** [comment]
+1

File: include/erlcloud_aws.hrl [hunk 1]
```diff
@@ -37,8 +37,8 @@
           autoscaling_scheme="https://"::string(),
           autoscaling_host="autoscaling.us-east-1.amazonaws.com"::string(),
           autoscaling_port=80::non_neg_integer(),
-          access_key_id::string()|undefined|false,
-          secret_access_key::string()|undefined|false,
+          access_key_id::string()|undefined,
```
Comment (zacharyfox) [inline include/erlcloud_aws.hrl:40]: Maybe better to update this to always a string with a default value of ""?

File: src/erlcloud_aws.erl [hunk 2]
```diff
@@ -208,7 +208,7 @@ default_config() ->
 
 -spec update_config(aws_config()) -> {ok, aws_config()} | {error, term()}.
 update_config(#aws_config{access_key_id = KeyId} = Config)
-  when is_list(KeyId) ->
+  when is_list(KeyId), KeyId /= [] ->
```
Comment (zacharyfox) [inline src/erlcloud_aws.erl:211]: Why not pattern match in function head?

File: include/erlcloud_aws.hrl (hunk 1 above)
Comment (motobob) [inline include/erlcloud_aws.hrl:40]: yeah...will do it in together with any next PR

**Pull request merged**